src/ui/state/history.rs

```diff
@@ -59,20 +56,17 @@ impl HistoryState {
             }
         }
 
-        return Command::perform(
-            get_onchain_txs(self.revaultd.clone(), outpoint),
-            Message::VaultOnChainTransactions,
-        );
-    }
-
-    pub fn update_selected_vault(&mut self, vault_txs: VaultTransactions) {
         if let Some(i) = self
             .vaults
             .iter()
-            .position(|vlt| vlt.vault.outpoint() == vault_txs.vault_outpoint)
+            .position(|vlt| vlt.vault.outpoint() == outpoint)
         {
-            self.selected_vault = Some(SelectedVault::new(self.vaults[i].vault.clone(), vault_txs));
+            let selected_vault = SelectedVault::new(self.vaults[i].vault.clone());
```

Comment:
nit: I think it would be cleaner using `find`:
```
if let Some(selected) = self
    .vaults
    .iter()
    .find(|vlt| vlt.vault.outpoint() == outpoint)
{
    let selected_vault = SelectedVault::new(selected.vault.clone());
    let cmd = selected_vault.load(self.revaultd.clone());
    self.selected_vault = Some(selected_vault);
    return cmd;
};
```

Reply:
thanks, fixed